### DISCUSSION TITLE
Svelte store and types

### DISCUSSION THREAD

**Discussion by lkj4** [sveltejs/kit#6150] [Q&A]
Why do I not get an error message on the last line? The property `kfjsdlf` doesn't exist in `test`.
```typescript
import { writable } from "svelte/store";

const test = {
  a: {
    b: 30,
    c: 20
  },
  d: true
};

const store = writable(test);

store.update((s) => ({ ...s, kfjsdlf: 0 }));
```
https://codesandbox.io/s/solitary-leftpad-wjh0py?file=/src/index.ts

**Comment by MilesWellsSVT** [comment]
If you check the type for the first argument of `update` you'll see it's `type Updater<T> = (value: T) => T;`

[Since TS 3.5 the implicit type constraint](https://github.com/microsoft/TypeScript-wiki/blob/23a53b8b7a9f0be7c12beab241fd1423a2336898/Breaking-Changes.md#typescript-35) for generics is `unknown`. This is why you can expand the type of the object you're returning. 

AFAIK this has to do with the limitations of the TS compiler's ability to infer generic typings.
